java/memory/src/main/java/org/apache/arrow/memory/BufferLedger.java

```diff
@@ -523,7 +522,7 @@ void print(StringBuilder sb, int indent, BaseAllocator.Verbosity verbosity) {
     }
   }
 
-  public UnsafeDirectLittleEndian getUnderlying() {
-    return allocationManager.getMemoryChunk();
+  public AllocationManager getAllocationManager() {
```

Comment:
This is breaking but the method seems to be used only in test. Any suggestions?

Reply:
Let's add a new method which is 
T unwrap(Class<T> clazz) 

Then mark getUnderlying as deprecated and have it delegatee to unwrap(UnsafeDirectLittleEndian.class) and then also support unwrap(AllocationManager.class).